zerver/views/auth.py

```diff
@@ -611,17 +616,18 @@ def remote_user_jwt(request: HttpRequest, token: str = REQ(default="")) -> HttpR
     return login_or_register_remote_user(request, result)
 
 
-@has_request_variables
+@typed_endpoint
 def oauth_redirect_to_root(
     request: HttpRequest,
+    *,
     url: str,
     sso_type: str,
     is_signup: bool = False,
     extra_url_params: Mapping[str, str] = {},
-    next: str | None = REQ(default=None),
-    multiuse_object_key: str = REQ(default=""),
-    mobile_flow_otp: str | None = REQ(default=None),
-    desktop_flow_otp: str | None = REQ(default=None),
+    next: str | None = None,
+    multiuse_object_key: str = "",
+    mobile_flow_otp: str | None = None,
+    desktop_flow_otp: str | None = None,
```

Comment:
I'm pretty sure this conversion would have introduced a security bug; the above parameters would incorrectly become passable via the API.

``` diff
diff --git a/zerver/views/auth.py b/zerver/views/auth.py
index f4c393e36f..611928a9b9 100644
--- a/zerver/views/auth.py
+++ b/zerver/views/auth.py
@@ -565,7 +565,6 @@ def remote_user_sso(
     return login_or_register_remote_user(request, result)
 
 
-@typed_endpoint
 def get_email_and_realm_from_jwt_authentication_request(
     request: HttpRequest, *, json_web_token: str
 ) -> tuple[str, Realm]:
@@ -619,11 +618,15 @@ def remote_user_jwt(request: HttpRequest, *, token: str = "") -> HttpResponse:
 @typed_endpoint
 def oauth_redirect_to_root(
     request: HttpRequest,
-    *,
     url: str,
     sso_type: str,
-    is_signup: bool = False,
-    extra_url_params: Mapping[str, str] = {},
+    is_signup: bool,
+    extra_url_params: Mapping[str, str],
+    # Protect the above parameters from being processed as kwargs
+    # provided by @typed_endpoint by marking them as mandatory
+    # positional parameters.
+    /,
+    *,
     next: str | None = None,
     multiuse_object_key: str = "",
     mobile_flow_otp: str | None = None,
@@ -723,7 +726,11 @@ def start_social_login(
             return config_error(request, backend)
 
     return oauth_redirect_to_root(
-        request, url=backend_url, sso_type="social", extra_url_params=extra_url_params
+        request,
+        backend_url,
+        "social",
+        False,
+        extra_url_params
     )
 
 
@@ -747,10 +754,10 @@ def start_social_signup(
         extra_url_params = {"idp": extra_arg}
     return oauth_redirect_to_root(
         request,
-        url=backend_url,
-        sso_type="social",
-        is_signup=True,
-        extra_url_params=extra_url_params,
+        backend_url,
+        "social",
+        True,
+        extra_url_params,
     )
 
 
```